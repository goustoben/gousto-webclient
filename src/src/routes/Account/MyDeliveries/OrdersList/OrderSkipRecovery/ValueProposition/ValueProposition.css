.container {
	margin: 30px 0;
}

.title {
	font-size: 16px;
	font-weight: 500;
	line-height: 16px;
	text-align: center;
	margin: 10px auto;
	font-family: Avenir;
}

.message{
	font-size: 14px;
	font-weight: 300;
	line-height: 19px;
	text-align: center;
	font-family: Avenir;
}
