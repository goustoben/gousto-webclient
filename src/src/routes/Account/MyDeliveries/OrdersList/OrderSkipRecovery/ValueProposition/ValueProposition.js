import React, { PropTypes } from 'react'
import { ModalContent } from 'ModalComponent'

import css from './ValueProposition.css'

const propTypes = {
	featureFlag: PropTypes.boolean,
	valueProposition: PropTypes.shape({
		title: PropTypes.string.isRequired,
		message: PropTypes.string.isRequired,
	})
}

const defaultProps = {
	valueProposition: {
		title: 'Not in on your delivery date?',
		message: 'Change your delivery day easily for any box you can choose recipes for.',
	}
}

const ValueProposition = ({ featureFlag, valueProposition }) => (
	(featureFlag)
		? (
			<ModalContent>
				<div className={css.title}>{valueProposition.title}</div>
				<div className={css.message}>{valueProposition.message}</div>
			</ModalContent>
		) : null
)

ValueProposition.propTypes = propTypes

ValueProposition.defaultProps = defaultProps

export default ValueProposition
