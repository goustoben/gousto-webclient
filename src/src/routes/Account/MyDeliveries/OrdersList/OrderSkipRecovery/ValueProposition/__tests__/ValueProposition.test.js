import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import ValueProposition from '../ValueProposition'

describe('Order Skip Modal Value Proposition', () => {
  let wrapper

  describe('Initial Render', () => {
    test('should render snapshot', () => {
      const tree = renderer.create(
				<ValueProposition />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })

    test('should render value proposition title', () => {
      wrapper = mount(<ValueProposition />)

      const title = wrapper.find('.title')

      expect(title.text()).toBe('Not in on your delivery date?')
    })

    test('should render value proposition message', () => {
      wrapper = mount(<ValueProposition />)

      const message = wrapper.find('.message')

      expect(message.text()).toBe('Change your delivery day easily for any box you can choose recipes for.')
    })
  })

  describe('Alternative Render', () => {
    test('should be able to render custom value propositionn message', () => {
      const valueProposition = {
        title: 'test title',
        message: 'test message'
      }

      wrapper = mount(<ValueProposition valueProposition={valueProposition} />)

      const title = wrapper.find('.title')
      const message = wrapper.find('.message')

      expect(title.text()).toBe('test title')
      expect(message.text()).toBe('test message')
    })
  })
})
