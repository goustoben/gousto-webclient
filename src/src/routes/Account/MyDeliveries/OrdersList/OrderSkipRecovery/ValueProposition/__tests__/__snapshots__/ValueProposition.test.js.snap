// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Skip Modal Value Proposition Initial Render should render snapshot 1`] = `
<div
  className="modalContent"
>
  <div
    className="title"
  >
    Not in on your delivery date?
  </div>
  <div
    className="message"
  >
    Change your delivery day easily for any box you can choose recipes for.
  </div>
</div>
`;
